app/assets/javascripts/discourse/app/components/sidebar/toggle-all-sections.gjs

```diff
@@ -0,0 +1,54 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import { ADMIN_NAV_MAP } from "discourse/lib/sidebar/admin-nav-map";
+
+export default class ToggleAllSections extends Component {
+  @service sidebarState;
+  @service keyValueStore;
+  @tracked collapsedSections = this.sidebarState.collapsedSections;
+
+  get allSectionsExpanded() {
+    return ADMIN_NAV_MAP.every((adminNav) => {
+      const collapsedSidebarSectionKey = `sidebar-section-${this.sidebarState.currentPanel.key}-${adminNav.name}-collapsed`;
+      return !this.collapsedSections.includes(collapsedSidebarSectionKey);
+    });
+  }
+
+  get title() {
+    return this.allSectionsExpanded
+      ? "admin.collapse_all_sections"
+      : "admin.expand_all_sections";
+  }
+
+  get icon() {
+    return this.allSectionsExpanded
+      ? "discourse-chevron-collapse"
+      : "discourse-chevron-expand";
+  }
+
+  @action
+  toggleAllSections() {
+    const collapseOrExpand = this.allSectionsExpanded
+      ? this.sidebarState.collapseSection.bind(this)
```

Comment:
How come `bind(this)` is needed here? Doesn't this override the `this` inside sidebar-state?

Reply:
yeah, without because we assign a function to the variable we lose the context of this. Without `bind` it we have error

<img width="628" alt="Screenshot 2024-03-27 at 9 41 05 AM" src="https://github.com/discourse/discourse/assets/72780/43e81810-553f-4ba8-8e9c-78979c6e5357">

instead of assigning collapseOrExpand, I can change it to if/else if you prefer.

Reply:
Nah all good leave it as is, it's fine :+1: